tests/test_unit/test_kinematics.py

```diff
@@ -182,3 +184,120 @@ def test_approximate_derivative_with_invalid_order(order):
+            ],
+            dims=["time", "individuals", "keypoints", "space"],
+            coords={
+                "time": time,
+                "individuals": individuals,
+                "keypoints": keypoints,
+                "space": space,
+            },
+        )
+        return ds
+
+    @pytest.fixture
+    def mock_dataarray_3D(self):
+        """Return a 3D DataArray containing a known head orientation."""
+        time = [0]
+        individuals = ["individual_0"]
+        keypoints = ["left", "right"]
+        space = ["x", "y", "z"]
+
+        ds = xr.DataArray(
+            [
+                [[[1, 0, 0], [-1, 0, 0]]],
+            ],
+            dims=["time", "individuals", "keypoints", "space"],
+            coords={
+                "time": time,
+                "individuals": individuals,
+                "keypoints": keypoints,
+                "space": space,
+            },
+        )
+        return ds
+
+    def test_compute_head_direction_vector(
+        self, mock_dataarray, mock_dataarray_3D
+    ):
+        """Test that the correct head direction vectors
+        are computed from a basic mock dataset.
+        """
+        # Test that validators work
+
+        # Catch incorrect datatype
+        with pytest.raises(TypeError, match="must be an xarray.DataArray"):
+            kinematics.compute_head_direction_vector(
+                mock_dataarray.values, "left_ear", "right_ear"
+            )
+
+        # Catch incorrect dimensions
+        with pytest.raises(
+            AttributeError, match="'time', 'space', and 'keypoints'"
+        ):
+            mock_data_keypoint = mock_dataarray.sel(
+                keypoints="nose", drop=True
+            )
+            kinematics.compute_head_direction_vector(
+                mock_data_keypoint, "left_ear", "right_ear"
+            )
+
+        # Catch identical left and right keypoints
+        with pytest.raises(ValueError, match="keypoints may not be identical"):
+            kinematics.compute_head_direction_vector(
+                mock_dataarray, "left_ear", "left_ear"
+            )
+
+        # Catch incorrect spatial dimensions
+        with pytest.raises(
+            ValueError,
+            match=re.escape("must have 2 (and only 2) spatial dimensions"),
+        ):
+            kinematics.compute_head_direction_vector(
+                mock_dataarray_3D, "left", "right"
+            )
+
+        # Test that output contains correct datatype, dimensions, and values
+        head_vector = kinematics.compute_head_direction_vector(
+            mock_dataarray, "left_ear", "right_ear"
+        )
+        known_vectors = np.array([[[0, 2]], [[-2, 0]], [[0, -2]], [[2, 0]]])
+
+        assert (
+            isinstance(head_vector, xr.DataArray)
+            and ("space" in head_vector.dims)
+            and ("keypoints" not in head_vector.dims)
+        )
+        assert np.equal(head_vector.values, known_vectors).all()
```

Comment:
This feels like a separate test too (the one for the `valid_dataarray` fixture maybe?)